app/controllers/api/base_controller.rb

```diff
@@ -1,4 +1,6 @@
 class Api::BaseController < ApplicationController
+  before_action :set_cache_headers
```

Comment:
It would be good to add a test that ensures that we have the headers in place the first time. We can trust then that the framework will support caching in subsequent requests.

Something simple like:

```ruby  
it "set an ETag header" do
    response.headers['ETag'].should be_present
  end
it "sets 'Last-Modified'" do
    response.headers['Last-Modified'].should be_present
end 
```

Reply:
I don't think this test is valuable. I would expect it to pass even without the code I just put in?

Reply:
Sorry I didn’t try the code myself. I just thought that it would work as the headers shouldn’t exist unless we set them explicitly.  I find value in being explicit about having a test that give us confidence that our cache is in place. Can you think of a better way?

Reply:
Sorry I didn’t try the code myself. I just thought that it would work as the headers shouldn’t exist unless we set them explicitly.  I find value in being explicit about having a test that give us confidence that our cache is in place. Can you think of a better way?